src/Symfony/Bundle/FrameworkBundle/Command/WorkerAmqpMoveCommand.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class WorkerAmqpMoveCommand extends ContainerAwareCommand
+{
+    private $broker;
+    private $logger;
+
+    protected function configure()
+    {
+        $this
+            ->setName('worker:amqp:move')
+            ->setDescription('Take all messages from a queue, and send them to the default exchange with a new routing key.')
+            ->setDefinition(array(
+                new InputArgument('from', InputArgument::REQUIRED, 'The queue.'),
+                new InputArgument('to', InputArgument::REQUIRED, 'The new routing key.'),
+            ))
+        ;
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->broker = $this->getContainer()->get('amqp.broker');
+        $this->logger = $this->getContainer()->get('logger');
```

Comment:
Yeah, command as a service seems to be the pattern here, except that it locates the services on initialization by itself